src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -0,0 +1,217 @@
+    put_bucket_tagging: { regular: "s3:PutBucketTagging" },
+    put_bucket_versioning: { regular: "s3:PutBucketVersioning" },
+    put_bucket_website: { regular: "s3:PutBucketWebsite" },
+    put_bucket_object_lock: { regular: "s3:PutBucketObjectLockConfiguration" },
+    put_bucket: { regular: "s3:CreateBucket" },
+    put_object_acl: { regular: "s3:PutObjectAcl" },
+    put_object_tagging: { regular: "s3:PutObjectTagging", versioned: "s3:PutObjectVersionTagging" },
+    put_object_uploadId: { regular: "s3:PutObject" },
+    put_object_retention: { regular: "s3:PutObjectRetention" },
+    put_object_legal_hold: { regular: "s3:GetObjectLegalHold"},
+    put_object: { regular: "s3:PutObject" },
+});
+
+const qm_regex = /\?/g;
+const ar_regex = /\*/g;
+
+const predicate_map = {
+    'StringEquals': (request_value, policy_value) => request_value === policy_value,
+    'StringNotEquals': (request_val, policy_value) => request_val !== policy_value,
+    'StringEqualsIgnoreCase': (request_val, policy_value) => request_val.toLowerCase() === policy_value.toLowerCase(),
+    'StringNotEqualsIgnoreCase': (request_val, policy_value) => request_val.toLowerCase() !== policy_value.toLowerCase(),
+    'StringLike': function(request_val, policy_value) {
+        const value_regex = RegExp(`^${policy_value.replace(qm_regex, '.?').replace(ar_regex, '.*')}$`);
+        return value_regex.test(request_val);
+    },
+    'StringNotLike': function(request_val, policy_value) {
+        const value_regex = RegExp(`^${policy_value.replace(qm_regex, '.?').replace(ar_regex, '.*')}$`);
+        return !value_regex.test(request_val);
+    },
+    'Null': function(request_value, policy_value) { return policy_value === 'true' ? request_value === null : request_value !== null; },
+};
+
+const condition_fit_functions = {
+    's3:ExistingObjectTag': _is_object_tag_fit,
+    's3:x-amz-server-side-encryption': _is_server_side_encryption_fit
+};
+
+const supported_actions = {
+    's3:ExistingObjectTag': ['s3:DeleteObjectTagging', 's3:DeleteObjectVersionTagging', 's3:GetObject', 's3:GetObjectAcl', 's3:GetObjectTagging', 's3:GetObjectVersion', 's3:GetObjectVersionTagging', 's3:PutObjectAcl', 's3:PutObjectTagging', 's3:PutObjectVersionTagging'],
+    's3:x-amz-server-side-encryption': ['s3:PutObject']
+};
+
+async function _is_server_side_encryption_fit(req, predicate, value) {
+    const encryption = s3_utils.parse_encryption(req);
+    const algorithm = encryption ? encryption.algorithm : null;
+    dbg.log1('bucket_policy: encrytpion fit?', value, algorithm, predicate(algorithm, value));
+    return predicate(algorithm, value);
+}
+
+async function _is_object_tag_fit(req, predicate, value) {
+    const reply = await req.object_sdk.get_object_tagging(req.params);
+    const tag = reply?.tagging?.find(element => (element.key === value.key));
+    const tag_value = tag ? tag.value : null;
+    dbg.log1('bucket_policy: object tag fit?', value, tag);
```

Comment:
add here the predicate response like you did in the previous function (again without calling it twice)